core/src/main/java/feast/core/service/SpecService.java

```diff
@@ -72,6 +77,59 @@ public SpecService(
     this.defaultSource = defaultSource;
   }
 
+  /**
+   * Get a feature set matching the feature name and version provided in the filter. The name
+   * is required. If the version is provided then it will be used for the lookup. If the version
+   * is omitted then the latest version will be returned.
+   *
+   * @param GetFeatureSetRequest containing the name and version of the feature set
+   * @return GetFeatureSetResponse containing a single feature set
+   */
+  public GetFeatureSetResponse getFeatureSet(GetFeatureSetRequest request)
+      throws InvalidProtocolBufferException {
+
+      // Validate input arguments
+      checkValidCharacters(request.getName(), "featureSetName");
+      if (request.getName().isEmpty()) {
+        throw io.grpc.Status.INVALID_ARGUMENT
+            .withDescription("No feature set name provided")
+            .asRuntimeException();
+      }
+      if (request.getVersion() < 0){
+        throw new IllegalArgumentException("Version number cannot be less than 0");
```

Comment:
I think we should throw standard GRPC error because `IllegalArgumentException` will not include useful GRPC error codes and details.

```suggestion
        throw io.grpc.Status.INVALID_ARGUMENT
            .withDescription("Version number cannot be less than 0")
            .asRuntimeException();
```

Reply:
Good catch